### PR TITLE
Fix the message for Make Field Readonly

### LINKED ISSUE #26219: Make Readonly doesn't have a description for the Error List
image

It shows Description as blank for IDE0044.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #26219

### Customer scenario

A customer upgrades to version 15.7. IDE0044 warnings start appearing, but no description is given for the warning so it's not clear to users what the problem is.

### Bugs this fixes

Fixes #26219

### Workarounds, if any

None, but also none needed. In order to understand a workaround, users would need to understand what IDE0044 is. Once a user understands what IDE0044 is without a description, no workaround is needed.

### Risk

Very low. The fix is production code is clear and concise.

### Performance impact

None.

### Is this a regression from a previous update?

IDE0044 is new for the 15.7 release, so "errors appear without a description" could be considered a regression. On the other hand, IDE0044 never shipped in a previous release with a description.

### Root cause analysis

Several items contributed to this issue:

* No generalized test was included for this behavior
* The IDE code is intentionally resilient to this type of failure, opting to omit missing text rather than fail (crash the IDE, etc.) because third-party extension developers can provide implementations of the same functionality

### How was the bug found?

Customer reported

### Test documentation updated?

No, but a regression test is added to ensure the situation does not reappear for existing or new diagnostics.

### CODE REVIEW

**Comment by sharwell** [comment]
@jinujoseph This is based on **dev15.7.x** branch so we can take it wherever needed.

File: src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs [hunk 1]
```diff
@@ -36,6 +36,102 @@ internal virtual (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderA
...
+            using (var workspace = new AdhocWorkspace())
+            {
+                DiagnosticAnalyzer diagnosticAnalyzer;
+                try
+                {
+                    diagnosticAnalyzer = CreateDiagnosticProviderAndFixer(workspace).Item1;
+                }
+                catch (NotSupportedException)
+                {
+                    return;
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs]: when does this happen?

File: src/Test/Utilities/Portable/Diagnostics/DescriptorFactory.cs [hunk 2]
```diff
@@ -14,15 +14,16 @@ public static class DescriptorFactory
...
+        /// - <see cref="DiagnosticDescriptor.MessageFormat"/> set to <paramref name="id"/>
         /// - <see cref="DiagnosticDescriptor.DefaultSeverity"/> set to <see cref="DiagnosticSeverity.Hidden"/>
         /// - <see cref="WellKnownDiagnosticTags.NotConfigurable"/> custom tag added in <see cref="DiagnosticDescriptor.CustomTags"/>.
         /// </remarks>
         /// <param name="id">The value for <see cref="DiagnosticDescriptor.Id"/>.</param>
         /// <returns>A <see cref="DiagnosticDescriptor"/> with specified <see cref="DiagnosticDescriptor.Id"/>.</returns>
         public static DiagnosticDescriptor CreateSimpleDescriptor(string id)
         {
-            return new DiagnosticDescriptor(id, title: "", messageFormat: "", category: "",
+            return new DiagnosticDescriptor(id, title: "", messageFormat: id, category: "",
```
Comment (CyrusNajmabadi) [inline src/Test/Utilities/Portable/Diagnostics/DescriptorFactory.cs]: is this appropriate?  seems... strange...

File: src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs (hunk 1 above)
Comment (sharwell) [reply src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs]: ➡️ 

https://github.com/dotnet/roslyn/blob/d95e0e6d31afba486804c301109f6b1ef2b6cbe2/src/EditorFeatures/VisualBasicTest/Diagnostics/AddImport/AddImportTests_NuGet.vb#L32-L34

File: src/Test/Utilities/Portable/Diagnostics/DescriptorFactory.cs (hunk 2 above)
Comment (sharwell) [reply src/Test/Utilities/Portable/Diagnostics/DescriptorFactory.cs]: 💭 It's easier than further work to specialize the behavior for a test-only analyzer.

File: src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs]: A comment might be wise here.

Comment (sharwell) [reply src/EditorFeatures/TestUtilities/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs]: ➡️ Fixed in 15f74696d039687953d4b36c452494fac7c06499

**Comment by jinujoseph** [comment]
Approved to merge for 15.8.Preview2

**Pull request merged**